lib/patches/active_record/quoting.rb

```diff
@@ -0,0 +1,21 @@
+if ::ActiveRecord::VERSION::STRING >= "5.0.0"
+  module Globalize
+    module ConnectionAdapters
+      module Quoting
+
+        private
+
+        def _quote(value)
+          case value
+            when Array      then
+              "'#{quote_string(YAML.dump(value))}'"
+            else
+              super
+          end
+        end
+      end
+    end
+  end
+
+  ActiveRecord::ConnectionAdapters::AbstractAdapter.include Globalize::ConnectionAdapters::Quoting
```

Comment:
I think that the issue about the `'_quote': can't  quote Array (TypeError)` is related to the test suite based only to the sqlite which doesn't support array datatypes. 
See: https://github.com/globalize/globalize/blob/master/test/data/schema.rb#L52

So I'm no sure that monkey patching the adapter is the right solution in this case.. But I need more time to investigate. :)